### DISCUSSION TITLE
Error IDX20803 after update to System.IdentityModel.Tokens.Jwt version 7.0.0

### DISCUSSION THREAD

**Discussion by MohammadMiras** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#2319] [Q&A]
i use **keycloak**  for authentication  updated the project packages After updating, I was faced with the error

`
‍‍‍‍ 07:37:19 - IDX20803: Unable to obtain configuration from: 'https://example.com/realms/Development/.well-known/openid-configuration'. IDX20803: Unable to obtain configuration from: 'https://example.com/realms/Development/.well-known/openid-configuration'. Could not load type 'Microsoft.IdentityModel.Json.JsonConvert' from assembly 'Microsoft.IdentityModel.Tokens, Version=7.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Could not load type 'Microsoft.IdentityModel.Json.JsonConvert' from assembly 'Microsoft.IdentityModel.Tokens, Version=7.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. => Microsoft.IdentityModel.Json.JsonConvert when communicating with example.com/realms/Development/.well-known/openid-configuration
`
‍‍.targets file information.
 `
 <PackageReference Include="Azure.Storage.Blobs" Version="12.17.0" />
    <PackageReference Include="Bogus" Version="34.0.2" />
    <PackageReference Include="ClosedXML" Version="0.102.1" />
    <PackageReference Include="Hashids.net" Version="1.7.0" />
    <PackageReference Include="Humanizer.Core" Version="2.14.1" />
    <PackageReference Include="Humanizer.Core.ar" Version="2.14.1" />
    <PackageReference Include="Humanizer.Core.fa" Version="2.14.1" />
    <PackageReference Include="Humanizer.Core.ru" Version="2.14.1" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="7.0.10" />
    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="7.0.4" />
    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="7.0.0" />
    <PackageReference Include="Microsoft.IdentityModel.Tokens" Version="6.32.3" />
    <PackageReference Include="morelinq" Version="3.4.2" />
    <PackageReference Include="MySqlConnector" Version="2.2.7" />
    <PackageReference Include="Pomelo.EntityFrameworkCore.MySql" Version="7.0.0" />
    <PackageReference Include="Pomelo.EntityFrameworkCore.MySql.NetTopologySuite" Version="7.0.0" />
    <PackageReference Include="SocketIOClient" Version="3.1.1" />
    <PackageReference Include="System.Data.SqlClient" Version="4.8.5" />
    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.32.3" />
    <PackageReference Include="System.Linq.Dynamic.Core" Version="1.3.4" />
    <PackageReference Include="System.Text.Json" Version="7.0.3" />
`
Add Authentication code  

```
services.AddAuthentication(options =>
        {
            options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
            options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
        })
        .AddJwtBearer(jwtOptions =>
        {
            jwtOptions.Authority = "https://example.com/realms/Development";
            jwtOptions.Audience = "account";
            jwtOptions.Events = new JwtBearerEvents()
            {
                OnAuthenticationFailed = c =>
                {
                    c.NoResult();

                    Logger.LogException(c.Exception);

                    c.Response.StatusCode = 500;
                    c.Response.ContentType = "text/plain";
                    if (ApiConfig.IsDeveloping)
                    {
                        return c.Response.WriteAsync(c.Exception.ToString());
                    }
                    return c.Response.WriteAsync("An error occured processing your authentication.");
                },
                OnTokenValidated = c =>
                {
                    if (c.Principal.IsInRole("SuperAdmin"))
                    {
                        c.HttpContext.Items["IsSuperAdmin"] = true;
                    }
                    return System.Threading.Tasks.Task.CompletedTask;
                }
            };
        });
```

Every request that is sent from the client to the api, OnAuthenticationFailed is called

**Comment by jmprieur** [comment]
Did you make sure.that you are referencing thr same version 7.0.9 for all the Microsoft.IdentittyModel assemblies?

Also are you on .net 8?

**Comment by pberggreen** [comment]
I get the same error when I update System.IdentityModel.Tokens.Jwt from version 6.34.0 to version 7.0.3.
I use the same version in all assemblies. 
I'm on net7.0.
